lib/helpers/UserGroupUiHelper.ts

```diff
@@ -30,37 +80,74 @@ export class UserGroupUiHelper extends UiBaseLocators {
 
   async clickUserGroupWithName(name: string) {
     await this.page.getByRole('link', {name: name}).click();
+    await this.page.waitForTimeout(200);
   }
 
-  async clickPermissionByName(permissionName: string[]) {
+  async clickPermissionsByName(permissionName: string[]) {
     for (let i = 0; i < permissionName.length; i++) {
       await this.permissionVerbBtn.getByText(permissionName[i], {exact: true}).click();
     }
   }
 
+  async clickGranularPermissionsByName(permissionName: string[]) {
+    for (let i = 0; i < permissionName.length; i++) {
+      await this.granularPermissionsModal.getByText(permissionName[i], {exact: true}).click();
+    }
+  }
+
   async doesUserGroupHavePermission(permissionName: string, hasPermission = true) {
-    return await expect(this.permissionVerbBtn.filter({has: this.page.getByText(permissionName, {exact: true})}).locator('uui-toggle').locator('#icon-checked').getByRole('img')).toBeVisible({visible: hasPermission});
+    await expect(this.permissionVerbBtn.filter({has: this.page.getByLabel(permissionName, {exact: true})}).locator('uui-toggle').locator('#icon-checked').getByRole('img')).toBeVisible({visible: hasPermission});
```

Comment:
locator('uui-toggle').locator('#icon-checked').getByRole('img')) - this should be defined in the construction